native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/channel/TargetChannel.java

```diff
@@ -165,4 +167,20 @@ public boolean isRequestHeaderWritten() {
     public HttpRoute getHttpRoute() {
         return httpRoute;
     }
+
+    public void setPool(GenericObjectPool pool) {
+        this.pool = pool;
+    }
+
+    public void invalidate() {
+        if (pool != null) {
+            try {
+                pool.invalidateObject(this);
+            } catch (Exception e) {
+                LOG.error("Error while invalidating the channel: {}", this, e);
+            }
+        } else {
+            LOG.warn("Pool is not set for the channel: {}", this);
+        }
```

Comment:
I think this might look cleaner, but it is a subjective take.
```suggestion
        if (pool == null) {
            LOG.warn("Pool is not set for the channel: {}", this);
            return;
        }
        try {
            pool.invalidateObject(this);
        } catch (Exception e) {
            LOG.error("Error while invalidating the channel: {}", this, e);
        }
```

Reply:
Fixed via [79d8de8](https://github.com/ballerina-platform/module-ballerina-http/pull/2448/commits/79d8de8a5fa597a8449705842fcc9a232f0ff95d)